src/strace.c

```diff
@@ -3067,7 +3070,22 @@ cleanup(int fatal_sig)
 			kill(tcp->pid, SIGCONT);
 			kill(tcp->pid, fatal_sig);
 		}
-		detach(tcp);
+		detach(tcp, true);
+		if (tcp->pid)
+			cnt++;
+	}
+	while (cnt) {
+		int status;
+		pid_t pid = waitpid(-1, &status, __WALL);
+		debug_func_msg("looking at pid %u", pid);
+		struct tcb *tcp = pid2tcb(pid);
+		if (!tcp || !tcp->pid) {
+			int error = ptrace(PTRACE_DETACH, pid, 0, 0);
+			debug_func_msg("ptrace untraced: %u, %x %d", pid, status, error);
+			continue;
+		}
```

Comment:
Invoking PTRACE_DETACH on a process which is not our tracee doesn't look correct.

Reply:
for the golang test process, the grandchild process will get here, which may be caused by invoking`detach` twice. the new commited code is ok and `PTRACE_DETACH` was deleted.